src/libraries/System.Net.Mail/src/System/Net/Mail/SmtpClient.cs

```diff
@@ -1092,8 +1092,10 @@ protected virtual void Dispose(bool disposing)
                     _cancelled = true;
                     Abort();
                 }
-
-                _transport?.ReleaseConnection();
+                else
+                {
+                    _transport?.ReleaseConnection();
```

Comment:
@scalablecory @tmds in this way the path should be clear, if we're incall we'll abort without quit otherwise cleanup with quit.